tests/TestCase/ORM/TableTest.php

```diff
@@ -511,6 +511,29 @@ function (Event $event, $query, $options) use ($expected) {
     }
 
     /**
+     * Test that the association() method supports the dot syntax.
+     *
+     * @return void
+     */
+    public function testAssociationDotSyntax()
+    {
+        $groups = TableRegistry::get('Groups');
+        $members = TableRegistry::get('Members');
+        $groupsMembers = TableRegistry::get('GroupsMembers');
+
+        $groups->belongsToMany('Members');
+        $groups->hasMany('GroupsMembers');
+        $groupsMembers->belongsTo('Members');
+
+        $association = $groups->association('GroupsMembers.Members');
```

Comment:
Could we get a test for associations deeper than 2 levels.

Reply:
Definitely. Do you want me to crawl the same associations I have set up there or mix it up?